Lib/shutil.py

```diff
@@ -73,8 +73,14 @@ class RegistryError(Exception):
     and unpacking registries fails"""
 
 
-def copyfileobj(fsrc, fdst, length=16*1024):
-    """copy data from file-like object fsrc to file-like object fdst"""
+def copyfileobj(fsrc, fdst, length=None):
+    """Copy data from file-like object `fsrc` to file-like object `fdst`.
+
+    An in-memory buffer size in bytes can be set with `length`; the default is
+    16 KiB.
+    """
+    if not length:
+        length = 16 * 1024
     while 1:
```

Comment:
I'm not sure that it's ok to use a loop if the length is negative. I suggest to have a special case for negative value calling read() (no parameter) only *once*.